### PR TITLE
test: perform the ldapsearch in integration tests

### PR DESCRIPTION
**Pull request by wood-push-melon**
This pull request is to resolve [IAM-1000](https://warthogs.atlassian.net/browse/IAM-1000).

[IAM-1000]: https://warthogs.atlassian.net/browse/IAM-1000?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Let me know if the test coverage is good enough. (or we want to test more cases)

### CODE REVIEW

File: tests/integration/conftest.py [hunk 1]
```diff
@@ -29,58 +44,130 @@ def extract_certificate_common_name(certificate: str) -> Optional[str]:
...
 
+@pytest_asyncio.fixture
+async def database_address(ops_test: OpsTest) -> str:
+    return await unit_address(ops_test, app_name=DB_APP)
 
-@pytest.fixture
-def unit_integration_data(ops_test: OpsTest) -> Callable:
-    return functools.partial(get_unit_integration_data, ops_test.model_name)
 
+@pytest_asyncio.fixture
```
Comment (BarcoMasile) [inline tests/integration/conftest.py:153]: suggestion: this could benefit from a "session" scope or at least "package" scope
unless I'm missing something

**Review by BarcoMasile — DISMISSED** [review]
LGTM +1 question and 1 suggestion

File: tox.ini [hunk 2]
```diff
@@ -65,8 +65,21 @@ commands =
...
     coverage report
 
+[testenv:build-prerequisites]
+description = Install necessary Linux packages for python dependencies
+allowlist_externals =
+    sudo
+    apt-get
+commands =
+    sudo apt-get update
+    sudo apt-get install -y python3-dev \
```
[the remark this answers is not in the document]
Comment (wood-push-melon) [reply tox.ini:75]: This one is necessary for `python-ldap` building. [Here](https://www.python-ldap.org/en/python-ldap-3.4.3/installing.html#debian) is more information.

File: tests/integration/conftest.py (hunk 1 above)
Comment (wood-push-melon) [reply tests/integration/conftest.py:153]: Yeah. I will add the scope.

**Review by shipperizer — APPROVED** [review]

**Review by BarcoMasile — APPROVED** [review]

File: tox.ini (hunk 2 above)
[the remark this answers is not in the document]
Comment (shipperizer) [reply tox.ini:75]: python3-dev didnt seem necessary for installing, but given it's for the testenv i wouldnt sweat it

**Pull request merged**